File: 02-09-2026/lab5_2.py
n = int(input("Enter the value of n: "))

vec1 = list(map(float, input("Enter the values of the first vector: ").replace(',', ' ').split()))
vec2 = list(map(float, input("Enter the values of the second vector: ").replace(',', ' ').split()))

dot_product = sum(v1 * v2 for v1, v2 in zip(vec1, vec2))

print(f"The dot product is {dot_product:g}.")

File: 02-09-2026/lab5_2.c
#include <stdio.h>

int main() {
    int n;
    printf("Enter the value of n: ");
    scanf("%d", &n);

    double vec1[n], vec2[n];

    printf("Enter the values of the first vector: ");
    for (int i = 0; i < n; i++) {
        scanf("%lf", &vec1[i]);
    }

    printf("Enter the values of the second vector: ");
    for (int i = 0; i < n; i++) {
        scanf("%lf", &vec2[i]);
    }

    double dot_product = 0.0;
    for (int i = 0; i < n; i++) {
        dot_product += vec1[i] * vec2[i];
    }

    printf("The dot product is %g.\n", dot_product);

    return 0;
}

